From e85856a290c46c3c02fc1196b02e3aa6f99c71cb Mon Sep 17 00:00:00 2001
From: Kumar Gala <galak@kernel.crashing.org>
Date: Wed, 10 Mar 2010 01:09:27 +0000
Subject: [PATCH 089/252] dpa: Convert to netops

netops is required in future versions of the kernel, so we might as well
switch to it now.

Signed-off-by: Kumar Gala <galak@kernel.crashing.org>
Signed-off-by: Andy Fleming <afleming@freescale.com>
[Original patch taken from Freescale p4080 SDK 2.1 ISO
image. Apply cleanly to kernel 2.6.34]
Integrated-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/dpa/dpa.c |   18 ++++++++++++------
 1 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/drivers/net/dpa/dpa.c b/drivers/net/dpa/dpa.c
index df821ca..4632e39 100644
--- a/drivers/net/dpa/dpa.c
+++ b/drivers/net/dpa/dpa.c
@@ -2128,6 +2128,17 @@ _return:
 }
 #endif /* CONFIG_FSL_QMAN_FQRANGE */
 
+static const struct net_device_ops dpa_netdev_ops = {
+	.ndo_open = dpa_start,
+	.ndo_start_xmit = dpa_tx,
+	.ndo_stop = dpa_stop,
+	.ndo_change_rx_flags = dpa_change_rx_flags,
+	.ndo_tx_timeout = dpa_timeout,
+	.ndo_get_stats = dpa_get_stats,
+	.ndo_set_mac_address = eth_mac_addr,
+	.ndo_validate_addr = eth_validate_addr,
+};
+
 static int __devinit __cold __attribute__((nonnull))
 dpa_init_probe(struct of_device *_of_dev)
 {
@@ -2773,16 +2784,11 @@ dpa_probe(struct of_device *_of_dev)
 			&tx_port_param);
 	}
 
+	net_dev->netdev_ops = &dpa_netdev_ops;
 	net_dev->features		|= DPA_NETIF_FEATURES;
-	net_dev->get_stats		 = dpa_get_stats;
 	SET_ETHTOOL_OPS(net_dev, &dpa_ethtool_ops);
 	net_dev->needed_headroom	 = DPA_BP_HEAD;
-	net_dev->hard_start_xmit	 = dpa_tx;
 	net_dev->watchdog_timeo		 = tx_timeout * HZ / 1000;
-	net_dev->open			 = dpa_start;
-	net_dev->stop			 = dpa_stop;
-	net_dev->tx_timeout		 = dpa_timeout;
-	net_dev->change_rx_flags	 = dpa_change_rx_flags;
 
 	_errno = register_netdev(net_dev);
 	if (unlikely(_errno < 0)) {
-- 
1.6.5.2

